encrypt.go

```diff
@@ -113,35 +112,30 @@ var ErrMaliciousData = sio.NotAuthentic
 // The data must be a valid ciphertext produced by
 // EncryptData. Otherwise, the decryption will fail.
 func DecryptData(password string, data io.Reader) ([]byte, error) {
-	var (
-		salt  [32]byte
-		id    [1]byte
-		nonce [8]byte // This depends on the AEAD but both used ciphers have the same nonce length.
-	)
-
-	if _, err := io.ReadFull(data, salt[:]); err != nil {
-		return nil, err
-	}
-	if _, err := io.ReadFull(data, id[:]); err != nil {
-		return nil, err
-	}
-	if _, err := io.ReadFull(data, nonce[:]); err != nil {
+	// Parse the stream header
+	var hdr [41]byte
+	if _, err := io.ReadFull(data, hdr[:]); err != nil {
+		if err == io.EOF {
+			// Incomplete header, return malicious data
+			return nil, ErrMaliciousData
```

Comment:
```suggestion
			return nil, io.ErrUnexpectedEOF
```

Reply:
@klauspost I think malicious data is better here. The point is to return a more meangniful error to the user to understand what is going on.

Besides, malicious data should be acceptable as well when we encounter io.EOF specifically. When there is io.EOF, it means the stream is ended in an expected way, and if we did not receive enough data for the header, I think we can consider it

Reply:
Even io.ErrUnexpectedEOF is okay - we don't really ignore it anyways for this caller. We can treat that at the caller layer on what the meaning of this io.ErrUnexpectedEOF must be.